pkg/policy/mapstate.go

```diff
@@ -374,12 +443,12 @@ func (keys MapState) denyPreferredInsertWithChanges(newKey Key, newEntry MapStat
 					keys[k] = v
 				}
 			}
+			keys.addKeyWithChanges(newKey, newEntry, adds, deletes)
+			return
 		}
-		keys.addKeyWithChanges(newKey, newEntry, adds, deletes)
-		return
+
 	}
 	// branch for adding a new allow L3-L4
-
 	newKeyCpy := newKey
 	newKeyCpy.DestPort = 0
 	newKeyCpy.Nexthdr = 0
```

Comment:
The conditional check below checks whether there is an L3-only deny that covers the new entry, and skips in that case. However, I think that the new code on lines 423-426 already handle this case so we probably don't need this check any more?

Reply:
That checks only the deny rules; this is here for the allow rules as well.